### PR TITLE
fix: revert user-delegation sas token usage in VHD builds

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

/kind fix

**What this PR does / why we need it**:

this reverts the sas token generation strategy back towards using a hard-coded connection-string rather than user-delegation, as tokens created using user-delegation have a maximum expiration time of 7d which is too short

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/generate-vhd-publishing-info.sh [hunk 1]
```diff
@@ -32,23 +33,20 @@ else
...
     expiry_date=$(date +"%Y-%m-%dT00:00Z" -d "+7 day")
-    sas_token=$(az storage container generate-sas --account-name ${OUTPUT_STORAGE_ACCOUNT_NAME} --name ${OUTPUT_STORAGE_CONTAINER_NAME} --permissions lr --expiry ${expiry_date} --auth-mode login --as-user | tr -d '"')
+    sas_token=$(az storage container generate-sas --name ${OUTPUT_STORAGE_CONTAINER_NAME} --permissions lr --connection-string ${CLASSIC_SA_CONNECTION_STRING} --start ${start_date} --expiry ${expiry_date} | tr -d '"')
 else
-    [ -z "${CLASSIC_SA_CONNECTION_STRING}" ] && echo "CLASSIC_SA_CONNECTION_STRING should be set when generating publishing info without a system-assigned identity..." && exit 1
-    echo "generating traditional SAS token with CLASSIC_SA_CONNECTION_STRING..."
     # we still need to use the original connection string when not using a system-assigned identity on 1ES pools
     start_date=$(date +"%Y-%m-%dT00:00Z" -d "-1 day")
     expiry_date=$(date +"%Y-%m-%dT00:00Z" -d "+1 year")
-    sas_token=$(az storage container generate-sas --name vhds --permissions lr --connection-string ${CLASSIC_SA_CONNECTION_STRING} --start ${start_date} --expiry ${expiry_date} | tr -d '"')
```
Comment (alexeldeib) [inline vhdbuilder/packer/generate-vhd-publishing-info.sh:51]: ehhhh??

File: vhdbuilder/packer/generate-vhd-publishing-info.sh [hunk 2]
```diff
@@ -32,23 +33,20 @@ else
...
-if [[ "${identity_name}" == "systemAssignedIdentity" ]]; then
+echo "generating traditional SAS token with CLASSIC_SA_CONNECTION_STRING..."
+if [[ "${OS_NAME,,}" != "windows" ]]; then
     [ -z "${OUTPUT_STORAGE_ACCOUNT_NAME}" ] && echo "OUTPUT_STORAGE_ACCOUNT_NAME should be set when generating publishing with a system-assigned identity..." && exit 1
     [ -z "${OUTPUT_STORAGE_CONTAINER_NAME}" ] && echo "OUTPUT_STORAGE_CONTAINER_NAME should be set when generating publishing info with a system-assigned identity..." && exit 1
-    echo "using ${identity_name} to generate user-delegation SAS token..."
     echo "storage account name: ${OUTPUT_STORAGE_ACCOUNT_NAME}"
     echo "storage container name: ${OUTPUT_STORAGE_CONTAINER_NAME}"
     # max of 7 day expiration time when using user delegation SAS
     expiry_date=$(date +"%Y-%m-%dT00:00Z" -d "+7 day")
```
Comment (alexeldeib) [inline vhdbuilder/packer/generate-vhd-publishing-info.sh:43]: errrrr

File: vhdbuilder/packer/generate-vhd-publishing-info.sh (hunk 1 above)
Comment (cameronmeissner) [reply vhdbuilder/packer/generate-vhd-publishing-info.sh:51]: no one saw that

File: vhdbuilder/packer/generate-vhd-publishing-info.sh (hunk 2 above)
Comment (cameronmeissner) [reply vhdbuilder/packer/generate-vhd-publishing-info.sh]: or that

**Review by alexeldeib — APPROVED** [review]

**Comment by alexeldeib** [comment]
squash please :)

**Review by juan-lee — APPROVED** [review]

**Pull request merged**